crowbar_framework/app/helpers/nodes_helper.rb

```diff
@@ -88,16 +88,22 @@ def nodes_hash(group = nil)
     @nodes_hash[group] ||= begin
       {}.tap do |nodes|
         NodeObject.all.each do |node|
-          nodes[node.name] = {
-            :handle => node.handle,
-            :alias => node.alias,
-            :title => node.description(false, true),
-            :admin => node.admin?,
-            :group => node.group,
-            :platform => node[:platform],
-            :platform_version => node[:platform_version],
-            :cluster => false
-          } if node.group == group or group.nil?
+          if node.group == group || group.nil?
+            nodes[node.name] = {
+              :handle => node.handle,
+              :alias => node.alias,
+              :title => node.description(false, true),
+              :admin => node.admin?,
+              :group => node.group,
+              :platform => node[:platform],
+              :platform_version => node[:platform_version],
+              :cluster => false
+            }
+            if node["crowbar_wall"] && node["crowbar_wall"]["disk_roles"]
```

Comment:
Why does it not exist? It's iterating over the NodeObject, so this method have to exist and would result in:

```ruby
        NodeObject.all.each do |node|
          nodes[node.name] = {
            :handle => node.handle,
            :alias => node.alias,
            :title => node.description(false, true),
            :admin => node.admin?,
            :group => node.group,
            :platform => node[:platform],
            :platform_version => node[:platform_version],
            :cluster => false,
            :disk_roles => node.disk_roles
          } if node.group == group or group.nil?
```